### PR TITLE
DIV-4117 Petitioner can see the details of the DN approval in their account

### PR DESCRIPTION
**Pull request by ravinderpal**
# Description

Petitioner can see the details of the DN approval in their account

Fixes https://tools.hmcts.net/jira/browse/DIV-4117

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: app.js [hunk 1]
```diff
@@ -16,6 +17,19 @@ const getFilters = require('views/filters');
 
 const app = express();
 
+const middleware = [ idam.protect() ];
+
+const args = {
+  // string to define path to attach document donwload handler, default: '/document-download/:documentId' *optional*
+  uri: '/documents/:documentId',
+  // string to specifiy auth token cookie, default: '__auth-token' *optional*
+  authorizationTokenCookieName: '__auth-token',
```
Comment (dtaylor7) [inline app.js]: dont need

File: app.js [hunk 2]
```diff
@@ -16,6 +17,19 @@ const getFilters = require('views/filters');
 
 const app = express();
 
+const middleware = [ idam.protect() ];
+
+const args = {
+  // string to define path to attach document donwload handler, default: '/document-download/:documentId' *optional*
+  uri: '/documents/:documentId',
```
Comment (dtaylor7) [inline app.js]: dont need

File: app.js [hunk 3]
```diff
@@ -92,5 +99,6 @@ onePerPage.journey(app, {
   i18n: { filters: getFilters() },
   useCsrfToken: true
 });
+initDocumentHandler(app, middleware, args);
```
Comment (dtaylor7) [inline app.js]: can you remind me why this is after OPP init? Also can put all the config and definitions next to it

File: mocks/services/case-orchestration/retrieve-case/mock-case.json [hunk 4]
```diff
@@ -1,7 +1,7 @@
 {
   "caseId": "1234567890124567",
   "courts": "serviceCentre",
-  "state": "AwaitingDecreeNisi",
+  "state": "awaitingpronouncement",
```
Comment (dtaylor7) [inline mocks/services/case-orchestration/retrieve-case/mock-case.json]: mock state should be camelcase

File: steps/petition-progress-bar/PetitionProgressBar.step.js [hunk 5]
```diff
@@ -104,10 +108,16 @@ class PetitionProgressBar extends Interstitial {
     return this.case.permittedDecreeNisiReason ? this.case.permittedDecreeNisiReason : constants.undefendedReason;
   }
 
+  get isHearingDateExists() {
```
Comment (dtaylor7) [inline steps/petition-progress-bar/PetitionProgressBar.step.js]: makes more sense if its `doesHearingDateExists `

File: app.js (hunk 3 above)
Comment (dtaylor7) [reply app.js]: also will need to add some more config here soon, making changes to document handler now

File: services/documentHandler.js [hunk 6]
```diff
@@ -0,0 +1,20 @@
...
+
+const downloadDocumentEndpoint = '/documents';
+
+const documentWhiteList = config.document.documentWhiteList;
+
+const bind = app => {
+  const middleware = [ idam.protect() ];
+  const args = {
+    documentServiceUrl: `${config.services.evidenceManagementClient.url}${downloadDocumentEndpoint}`,
+    sessionFileCollectionsPaths: ['case.data.d8'],
```
Comment (alexwakeman) [inline services/documentHandler.js]: should use the config setting here

File: steps/petition-progress-bar/PetitionProgressBar.step.js [hunk 7]
```diff
@@ -31,6 +36,13 @@ class PetitionProgressBar extends Interstitial {
     return config.paths.petitionProgressBar;
   }
 
+  get isHearingDateInPast() {
+    const lastHearingDate = moment(
+      this.case.hearingDate[this.case.hearingDate.length - 1]
```
Comment (qzhou-hmcts) [inline steps/petition-progress-bar/PetitionProgressBar.step.js]: Nullpointer

File: config/default.yml [hunk 8]
```diff
@@ -29,7 +29,7 @@ services:
     useSSL: false
     encryptionAtRestKey: encryptionAtRestKey
   idam:
-    apiHealth: https://localhost:8001/health
+    apiHealth: https://localhost:8001/idam-api-health
```
Comment (tchow8) [inline config/default.yml]: If you run IDAM locally surely it runs on /health?

File: mocks/services/case-orchestration/retrieve-case/mock-case.json [hunk 9]
```diff
@@ -1,7 +1,7 @@
 {
   "caseId": "1234567890124567",
   "courts": "serviceCentre",
-  "state": "AwaitingDecreeNisi",
+  "state": "AwaitingPronouncement",
```
Comment (tchow8) [inline mocks/services/case-orchestration/retrieve-case/mock-case.json]: Why change this?

File: package.json [hunk 10]
```diff
@@ -29,12 +29,14 @@
   "pre-commit": "lint",
   "dependencies": {
     "@hmcts/div-idam-express-middleware": "^6.5.0",
-    "@hmcts/look-and-feel": "^2.1.1",
+    "@hmcts/div-document-express-handler": "^1.1.0",
+    "@hmcts/div-idam-express-middleware": "^6.4.0",
```
Comment (tchow8) [inline package.json]: duplicate of one above

File: package.json [hunk 11]
```diff
@@ -65,7 +67,7 @@
     "@hmcts/div-idam-test-harness": "^1.5.0",
     "@hmcts/div-test-harness": "https://github.com/hmcts/div-test-harness#a459aaa16a92cb13e0dac35c32bf559ad79de6a6",
     "@hmcts/node-js-environment-variable-setter": "https://github.com/hmcts/node-js-environment-variable-setter#0.2",
-    "@hmcts/one-per-page-test-suite": "^3.0.0",
+    "@hmcts/one-per-page-test-suite": "^3.0.2",
```
Comment (tchow8) [inline package.json]: Hmmm, not really part of this PR, but can probably remove one of the one-per-page-test-suite modules? Most probably the one in the dependencies section.

File: mocks/services/case-orchestration/retrieve-case/mock-case.json (hunk 9 above)
Comment (ravinderpal) [reply mocks/services/case-orchestration/retrieve-case/mock-case.json]: separate file added!

**Comment by hosainnet** [comment]
@ravinderpal quite a few conflicts..

**Comment by ravinderpal** [comment]
Closing this pr until ticket is unblocked
